src/librustc/ty/print/pretty.rs

```diff
@@ -1532,12 +1536,95 @@ define_print_and_forward_display! {
     }
 
     &'tcx ty::Const<'tcx> {
-        match self.val {
-            ConstValue::Unevaluated(..) |
-            ConstValue::Infer(..) => p!(write("_")),
-            ConstValue::Param(ParamConst { name, .. }) => p!(write("{}", name)),
-            _ => p!(write("{:?}", self)),
+        let u8 = cx.tcx().types.u8;
+        if let ty::FnDef(did, substs) = self.ty.sty {
+            p!(print_value_path(did, substs));
+            return Ok(cx);
+        }
+        if let ConstValue::Unevaluated(did, substs) = self.val {
+            match cx.tcx().describe_def(did) {
+                | Some(Def::Static(_, _))
+                | Some(Def::Const(_, false))
+                | Some(Def::AssociatedConst(_)) => p!(write("{}", cx.tcx().def_path_str(did))),
+                Some(Def::Const(_, true)) => p!(print_value_path(did, substs)),
```

Comment:
I think what was I was suggesting was to print `_` for this (and you can print `_` for anything not handled by consts and statics, so I'm not sure you need to change `describe_def`).

Reply:
done